pkg/datapath/connector/config.go

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package connector
+
+import (
+	"log/slog"
+	"math"
+
+	"github.com/cilium/hive/cell"
+
+	datapathOption "github.com/cilium/cilium/pkg/datapath/option"
+	"github.com/cilium/cilium/pkg/datapath/tunnel"
+	"github.com/cilium/cilium/pkg/datapath/types"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/option"
+	wgTypes "github.com/cilium/cilium/pkg/wireguard/types"
+)
+
+// Connector configuration. As per BIGTCP, the values here will not be calculated
+// until the Hive has started. This is necessary to allow other dependencies to
+// setup their interfaces etc.
+type ConnectorConfig struct {
+	// podDeviceHeadroom tracks the desired headroom buffer margin for the
+	// network device pair facing a workload.
+	podDeviceHeadroom uint16
+
+	// podDeviceTailroom tracks the desired tailroom buffer margin for the
+	// network device pairs facing a workload.
+	podDeviceTailroom uint16
+}
+
+func (cc *ConnectorConfig) GetPodDeviceHeadroom() uint16 {
+	return cc.podDeviceHeadroom
+}
+
+func (cc *ConnectorConfig) GetPodDeviceTailroom() uint16 {
+	return cc.podDeviceTailroom
+}
+
+type connectorParams struct {
+	cell.In
+
+	Lifecycle    cell.Lifecycle
+	Log          *slog.Logger
+	DaemonConfig *option.DaemonConfig
+	Orchestrator types.Orchestrator
+	WgAgent      wgTypes.WireguardAgent
+	TunnelConfig tunnel.Config
+}
+
+// Returns true if we should actively try and align the connector's netdev buffer
+// margins with that of the host's egress interfaces (e.g. tunnel, wireguard).
+func useTunedBufferMargins(datapathMode string) bool {
+	switch datapathMode {
+	case datapathOption.DatapathModeNetkit, datapathOption.DatapathModeNetkitL2:
+		return true
+	}
+	return false
+}
+
+// newConnectorConfig initialises a new ConnectorConfig object with default parameters.
+func newConfig(p connectorParams) (*ConnectorConfig, error) {
```

Comment:
```suggestion
func newConfig(p connectorParams) *ConnectorConfig {
```

I think we never return an error here